portal-web/test/functional/com/liferay/portalweb/macros/fragment/FragmentsEditor.macro

```diff
@@ -62,20 +62,47 @@ definition {
 
 		var fileExtension = File.getFileExtension(fileName = "${fileName}");
 
-		if ("${fileExtension}" == "html") {
-			FragmentsEditor.typeHtmlEditor(content = "${content}");
-		}
-		else if ("${fileExtension}" == "css") {
-			FragmentsEditor.typeCssEditor(content = "${content}");
-		}
-		else if ("${fileExtension}" == "js") {
-			FragmentsEditor.typeJsEditor(content = "${content}");
-		}
-		else if ("${fileExtension}" == "json") {
-			FragmentsEditor.typeConfigurationEditor(content = "${content}");
+		var browserType = PropsUtil.get("browser.type");
+
+		if ("${browserType}" == "internetexplorer") {
+			if ("${fileExtension}" == "html") {
+				Type.typeAceEditorNoError(
+					key_editorNumber = "1",
+					locator1 = "FragmentEditor#FRAGMENTS_EDITOR_IE",
+					value1 = "${content}");
+			}
+			else if ("${fileExtension}" == "css") {
+				Type.typeAceEditorNoError(
+					key_editorNumber = "2",
+					locator1 = "FragmentEditor#FRAGMENTS_EDITOR_IE",
+					value1 = "${content}");
+			}
+			else if ("${fileExtension}" == "js") {
+				Type.typeAceEditorNoError(
+					key_editorNumber = "3",
+					locator1 = "FragmentEditor#FRAGMENTS_EDITOR_IE",
+					value1 = "${content}");
+			}
+			else {
+				fail("Invalid file extension. Expected: html, css, js, or json. Got ${fileExtension}");
+			}
```

Comment:
So there wasn't a way to make the path compatible for both IE and other browsers?

Reply:
Yes, I checked.
The existing locators of fragment editor locate to //textarea which actually doesn't display on the UI, so on IE error "Element is not displayed" was thrown. But on Chrome, it can be found and type text in the content field simulating manually typing. You can see the below gif file.
![2019-11-27_18-23-04](https://user-images.githubusercontent.com/25576873/69715574-7dd7d800-1143-11ea-9cae-a893bed4324b.gif)
I am not completely sure if this is intended, but I think it's better to keep this on Chrome.
So I added a new locator instead of updating the existing ones even though the new locator also works for Chrome.

Reply:
got it.